### ISSUE TITLE
Saves not an option in versions to select from

### ISSUE THREAD

**Issue by Wolfkhan66** [umbraco/UmbracoDocs#3036]
***This is the page with issues: https://github.com/umbraco/UmbracoDocs/edit/main/Tutorials/Editors-Manual/Version-Management/Comparing-Versions/index.md***

Morning, In U7 you used to be able to select saves to rollback to, not just published versions. In U8 it now appears we can only select published versions to rollback to though the documentation still specifies 'all old versions of the page are saved to the history'. 

Is there now a config setting missing to enable this ability?

**Comment by sofietoft** [comment]
Thanks for reporting this @Wolfkhan66 !

I'll make sure to have someone with a little more knowledge around this give it a look as soon as possible.

**Comment by marcemarc** [comment]
@wolfkhan66 and @sofietoft 

It's not super clear in this documentation and I think we should perhaps update to be clearer... but it's also hard to explain:

When you open an Umbraco page to edit, and press Save, a version of it is created in Umbraco database. If you type some more, and press Save again, without leaving the page or publishing, then that 'in progress' version is just updated with your updates - an additional new version isn't created, so you can think of the version you are working on as 'your draft' - subsequent saves update the draft but don't create a new version - it isn't that granular (and that's not clear in the documentation) - if you publish the page, then that action creates another version, so if you press Save and then press Publish, you will have two versions! of the same content: the final saved version of your draft, and a new version that Publishing always creates... You should see both versions in 'rollback' with the same or very similar timestamp!

This is all fine and should mean nobody loses their work... in a single editor environment, saving as they go, they are always saving their work in progress.... But if after working for an hour, and saving every ten minutes, the editor would like to roll back to the version they had twenty minutes ago... they can't... only by publishing  at each stage would the draft be 'marked as complete' and a line to 'rollback to' created in the sand.

It can get more complicated if two editors are working on the same document at the same time without each other knowing... in this scenario, each editor will have their own draft version that gets saved to, independent of each other, the last person to save wins as to which version is considered to be the 'current' version... so if an editor is working on a draft, saves and goes for lunch, and comes back and revisits the page, they can be surprised to find their changes 'gone' and replaced with the other editors changes, which don't include the first editors work. Here it's possible to rollback to the editor's saved version, but at the expense of the changes the second editor made!!!

All very confusing, there used to be a core plugin,, Concierge that I think handled this kind of scenario with warnings. (and a package Look Who's Editing, and similar packages trying to address this problem... but I'm not sure if there is a 'winner' for V8)

There is a package from another 'woulfe'... called Preserver

https://our.umbraco.com/packages/backoffice-extensions/preserver-content-restoration/

that stores what the editor is entering in local storage that might be worth linking to on this page.

**Comment by Wolfkhan66** [comment]
@marcemarc Thank you for that breakdown Marc. We had this exact issue last week and it helped explain it to the team.
By the sounds of it what we use to have in U7 to select saves is actually custom work then (though I can't find it anywhere) so I will look into preserver as a option.

**Comment by Wolfkhan66** [comment]
@marcemarc is aware of this but I just wanted to reopen as it turns out what we have is not custom, and there is in fact a difference in Rollback versions in U7 and U8 that I think the documentation should reflect.

To try and summarize:
In U7 if a user saves (Without publishing) a new version is created that you can rollback to. Any further saves by that user update that version. 
If another user comes along and saves (without publishing) a brand new version is created that you can rollback to and any further saves by that user update that version. At this time, both of the users save versions can be rolled back to.

In U8 things change. If a user saves, there is no rollback version, as it is the current version it makes sense that you can't rollback and seems logical. However if another comes along and saves (without publishing) they take over that version instead of creating a new one and there is still nothing to rollback to since the last published version. In this situation whoever saves last wins. Only when a publish is performed do you get a version you can you rollback to. 

This seems like maybe it was done as an optimization, but rather we've found it has deepened potential data loss where multiple users might be editing a page at once without knowing (which in an ideal world shouldn't happen)

**Comment by marcemarc** [comment]
@Wolfkhan66 yes, trying to determine from HQ if this was a conscious design decision, eg better for performance, and multi editors same page is 'rare' / 'not supported' in which case we need to update the docs to make that clear or whether it was a scenario overlooked in V8, and the V7 behaviour is the desired scenario...
